src/main/java/com/mraof/minestuck/client/gui/ReadableSburbCodeScreen.java

```diff
@@ -0,0 +1,285 @@
+			RenderSystem.setShaderTexture(0, BOOK_TEXTURES_02);
+		else if(pageModulo == 2)
+			RenderSystem.setShaderTexture(0, BOOK_TEXTURES_03);
+		
+		
+		int topX = (this.width - GUI_WIDTH) / 2;
+		int topY = 2;
+		this.blit(poseStack, topX, topY, 0, 0, GUI_WIDTH, GUI_HEIGHT);
+		{
+			if(textList != null)
+			{
+				//pushPose function, followed by the scale function and ending with the popPose function, allow the scale changes to only be applied to the text
+				poseStack.pushPose();
+				float subtractScale = 0.4F;
+				float scale = (1 / subtractScale);
+				poseStack.scale(subtractScale, subtractScale, subtractScale);
+				
+				MutableInt lineY = new MutableInt();
+				
+				//takes the necessary page from the group of pages and then reads out each of the 40 lines stored for that page
+				for(String text : listOfPages.get(currentPage))
+				{
+					font.getSplitter().splitLines(text, TEXT_WIDTH, Style.EMPTY, true, (style, start, end) -> {
+						//limiting the length of the page via this if statement
+						if(stillValidLine(lineY.intValue()))
+						{
+							Component line = new TextComponent(text.substring(start, end)).setStyle(style);
+							font.draw(poseStack, line, ((this.width - GUI_WIDTH) / 2F + TEXT_OFFSET_X) * scale, (lineY.intValue() + TEXT_OFFSET_Y) * scale, 0x00A300); //hex is green
+							lineY.add(CUSTOM_LINE_HEIGHT);
+						}
+					});
+				}
+			}
+		}
+		
+		poseStack.popPose();
+		
+		super.render(poseStack, mouseX, mouseY, partialTicks);
+	}
+	
+	/**
+	 * Creates an array of booleans with each element representing a hieroglyph, elements are assigned true if the paradox code(first element) is present or a valid block contained in FULL_HIEROGLYPH_LIST is present
+	 */
+	private boolean[] checkForValidity(List<Block> recordedBlockList, boolean paradoxCode)
+	{
+		boolean[] booleans = new boolean[MAX_HIEROGLYPH_COUNT];
+		
+		for(int iterate = 0; iterate < MAX_HIEROGLYPH_COUNT; iterate++)
+		{
+			if(iterate == 0) //first iteration, checks for paradox code
+				booleans[iterate] = paradoxCode;
+			else if(!recordedBlockList.isEmpty() && iterate < FULL_HIEROGLYPH_LIST.size() + 1) //all but first iteration, checks for blocks in FULL_HIEROGLYPH_LIST, +1 to offset paradox code
+				booleans[iterate] = recordedBlockList.contains(FULL_HIEROGLYPH_LIST.get(iterate - 1));
+			else //unrecorded blocks
+				booleans[iterate] = false;
+		}
+		
+		return booleans;
+	}
+	
+	private boolean stillValidLine(int lineY)
+	{
+		return lineY < LINES_PER_PAGE * CUSTOM_LINE_HEIGHT;
+	}
+	
+	protected void createMenuControls()
+	{
+		if(this.minecraft != null)
+		{
+			this.addRenderableWidget(new Button(this.width / 2 - 100, 196, 200, 20, new TranslatableComponent("gui.done"), (p_214161_1_) -> {
+				this.minecraft.setScreen(null);
+			}));
+		}
+	}
+	
+	protected void createPageControlButtons()
+	{
+		int i = (this.width - 192) / 2;
+		this.forwardButton = this.addRenderableWidget(new PageButton(i + 116, 159, true, (p_214159_1_) -> {
```

Comment:
Don't use SRG-names in our own code. You should also use the lambda form without braces here.